src/java/org/apache/cassandra/service/accord/txn/TxnData.java

```diff
@@ -40,18 +41,77 @@
  * Fairly generic holder for result values for Accord txns as well as data exchange during Accord txn execution
  * when read results are returned to the coordinator to compute query results and writes.
  */
-public class TxnData extends HashMap<TxnDataName, TxnDataValue> implements TxnResult, Data
+public class TxnData extends Int2ObjectHashMap<TxnDataValue> implements TxnResult, Data
 {
     private static final long EMPTY_SIZE = ObjectSizes.measure(new TxnData());
 
+    private static final int TXN_DATA_NAME_INDEX_BITS = 32 - 6;
```

Comment:
why 6?

Reply:
Also, not important, but if we use the lower bits for kind we can encode as a `vint` (but, perhaps we don't really want to)

Reply:
This didn't need to count very high so I gave a few extra bits to support more kinds of data names in the future, but it's really not that necessary or helpful.